diff --git a/tests/test_file_proxy.py b/tests/test_file_proxy.py
index 2ec4d789e3..c73b23fe17 100644
--- a/tests/test_file_proxy.py
+++ b/tests/test_file_proxy.py
@@ -35,3 +35,20 @@ def test_new_lines():
     assert file.getvalue() == "-\n"
     file_proxy.flush()
     assert file.getvalue() == "-\n-\n"
+
+
+def test_isatty():
+    """Check isatty is proxied
+
+    Regression test for https://github.com/Textualize/rich/issues/4041
+
+    """
+
+    class TTYFile:
+        def isatty(self) -> bool:
+            return True
+
+    file = TTYFile()
+    console = Console()
+    file_proxy = FileProxy(console, file)
+    assert file_proxy.isatty()
