diff --git a/pydantic/json_schema.py b/pydantic/json_schema.py
index eb25c08..f2a100e 100644
--- a/pydantic/json_schema.py
+++ b/pydantic/json_schema.py
@@ -1756,13 +1756,16 @@ class GenerateJsonSchema:
         Returns:
             `True` if the field should be marked as required in the generated JSON schema, `False` otherwise.
         """
-        if self.mode == 'serialization' and self._config.json_schema_serialization_defaults_required:
-            return not field.get('serialization_exclude')
+        if self.mode == 'serialization':
+            if field.get('serialization_exclude_if') is not None:
+                return False
+            if self._config.json_schema_serialization_defaults_required:
+                return not field.get('serialization_exclude')
+
+        if field['type'] == 'typed-dict-field':
+            return field.get('required', total)
         else:
-            if field['type'] == 'typed-dict-field':
-                return field.get('required', total)
-            else:
-                return field['schema']['type'] != 'default'
+            return field['schema']['type'] != 'default'
 
     def dataclass_args_schema(self, schema: core_schema.DataclassArgsSchema) -> JsonSchemaValue:
         """Generates a JSON schema that matches a schema that defines a dataclass's constructor arguments.
