diff --git a/sympy/core/tests/test_power.py b/sympy/core/tests/test_power.py
index 80ae48c525c2..d0405e33cba5 100644
--- a/sympy/core/tests/test_power.py
+++ b/sympy/core/tests/test_power.py
@@ -668,3 +668,9 @@ def test_issue_25165():
     e1 = (1/sqrt(( - x + 1)**2 + (x - 0.23)**4)).series(x, 0, 2)
     e2 = 0.998603724830355 + 1.02004923189934*x + O(x**2)
     assert all_close(e1, e2)
+
+
+def test_issue_28219():
+    assert Pow(S.Zero, -1, evaluate=False).as_real_imag() == (S.NaN, S.NaN)
+    x = Symbol('x', real=True)
+    assert Pow(x, -1, evaluate=False).as_real_imag() == (1/x, S.Zero)
